src/lib.rs

```diff
@@ -129,7 +129,19 @@ impl NetworkResource {
     #[cfg(not(target_arch = "wasm32"))]
     pub fn listen(&mut self, socket_address: SocketAddr) {
         let mut server_socket = {
-            let socket = futures_lite::future::block_on(ServerSocket::listen(socket_address));
+            let webrtc_listen_addr = {
+                let webrtc_listen_ip: IpAddr = socket_address.ip();
+                let webrtc_listen_port = get_available_port(webrtc_listen_ip.to_string().as_str())
+                    .expect("no available port");
+
+                SocketAddr::new(webrtc_listen_ip, webrtc_listen_port)
+            };
+
+            let socket = futures_lite::future::block_on(ServerSocket::listen(
+                socket_address,
+                webrtc_listen_addr,
+                webrtc_listen_addr,
+            ));
```

Comment:
We should just make the `listen` function accept both `listen_address` and `public_address` - the same way it's now in `naia-socket`. Here's the link to the PR that introduced the change to `naia-socket`, which also describes the motivation behind it: https://github.com/naia-rs/naia-socket/pull/31.

Reply:
@mvlabat  Will do, I believe that would be a breaking change though. Would you like me to add documentation about using the solution that I used if you don't care about WebRTC listening ports?

Reply:
Pushed a new commit, do you think that the solution used is better?

Reply:
I think it's better, but I believe we should just have them as mandatory parameters. Making users think that just passing `0.0.0.0` as the `socket_addr` is enough will be quite counterintuitive for them. While it's fine for UDP, it'll never work for WebRTC.

Even if a user passes a value they'll get with `find_my_ip_address` to `socket_addr`, it won't work in production either: a socket will bind successfully, they'll be able to connect to the server locally, but remote connections won't work anyway. I think that we shouldn't give users a reason to think that some of the parameters are optional, because in production they'll need them all anyway, and it'll be a pretty difficult time for them to figure why it doesn't just work with the defaults when it worked locally.

So instead of providing convenient defaults that won't work, I think it would be better to educate users about providing correct arguments.

Reply:
Alright, I'll switch to that.

Reply:
Hi, this update mean Naia socket will start listen with webrtc and udp ports by default and user can't start a server with only udp socket?